packages/calcite-components/src/components/card/card.tsx

```diff
@@ -276,25 +277,14 @@ export class Card
         <calcite-checkbox
           checked={this.selected}
           label={this.messages.select}
-          onClick={this.selectCardDeprecated}
```

Comment:
This code was a bit problematic.
On click, calcite-card did `this.selected = !this.selected;`, which would flip the `checked` property in calcite-checkbox.
But, calcite-checkbox is also listening for `click` and would also flip the `checked` property in response.

This worked fine in Stencil because the order was as follows:
- calcite-card onClick fires (flipping property from true to false)
- calcite-checkbox onClick fires (flipping property from true to false)
- calcite-card re-renders, passing `checked: false` to calcite-checkbox, which is also the value `calcite-checkbox` set itself too

In Lit, the order is a bit different:
- calcite-card onClick fires (flipping property from true to false)
- calcite-card re-renders, passing `checked: false` to calcite-checkbox
- calcite-checkbox onClick fires, flipping property from false to true. Thus in effect we flipped the property twice, preventing the change.

Fixed by listening to `onCalciteCheckboxChange` instead and letting calcite-checkbox dictate the new post-flip state.